core/src/main/java/org/opensearch/sql/utils/DateTimeFormatters.java

```diff
@@ -133,6 +133,28 @@ public class DateTimeFormatters {
           .toFormatter(Locale.ROOT)
           .withResolverStyle(ResolverStyle.STRICT);
 
+  public static final DateTimeFormatter DATE_FORMATTER_VARIABLE_NANOS_OPTIONAL =
+      new DateTimeFormatterBuilder()
+          .appendPattern("[uu:MM:dd][uuuu-MM-dd HH:mm:ss][uuuu-MM-dd]")
```

Comment:
```suggestion
          .appendPattern("[HH:mm:ss][uuuu-MM-dd HH:mm:ss][uuuu-MM-dd]")
```
Please, add tests for all patterns.

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/134/commits/2016222f524af6c2edd56afa940a53181efd3b1c and